src/network_channel.c

```diff
@@ -9,14 +9,16 @@
 #ifdef NETWORK_CHANNEL_TCP_POSIX
 #include "platform/posix/tcp_ip_channel.c"
 #endif
-
 #elif defined(PLATFORM_RIOT)
 #ifdef NETWORK_CHANNEL_TCP_POSIX
 #include "platform/posix/tcp_ip_channel.c"
 #endif
 #ifdef NETWORK_CHANNEL_COAP_RIOT
 #include "platform/riot/coap_udp_ip_channel.c"
 #endif
+#ifdef NETWORK_CHANNEL_UART_RIOT
```

Comment:
I think here we should go with `NETWORK_CHANNEL_UART` only. The reason is that in the LF program we just specify the network channel type, and not the platform. So in the generated CMakeLists/Makefile we will add compile defs for the channel type, without knowing which platform we are compiling for yet.

The same goes for COAP.

Reply:
I agree